intel_nn_hal/Utils.h

```diff
@@ -271,11 +275,14 @@ class OpContainer {
             }
             if (i == opsVec.size() - 1) {
                 op = opsVec[i];
-                std::vector<uint32_t> ip_indices = op->getInputIndices();
-
-                op->setInputData(ip_indices[0], std::get<0>(intermediate_inp), std::get<1>(intermediate_inp));
-
-                op->run();
+                if (op->hasSubgraphInput()) {
```

Comment:
@anikulk Cant this be `!op->hasSubgraphInput()` with only the code in the `else `block?
And also `op->run();` can be just once.